Shared/API/CloudFunction.swift

```diff
@@ -11,11 +11,66 @@ import Parse
 import Combine
 
 protocol CloudFunction: StatusableRequest {
-    func makeRequest(andUpdate statusables: [Statusable], viewsToIgnore: [UIView]) -> AnyPublisher<ReturnType, Error>
+    func makeRequest(andUpdate statusables: [Statusable],
+                     viewsToIgnore: [UIView]) -> AnyPublisher<ReturnType, Error>
+    func makeAsyncRequest(andUpdate statusables: [Statusable],
+                          viewsToIgnore: [UIView]) async throws -> ReturnType
 }
 
 extension CloudFunction {
 
+    func makeAsyncRequest(andUpdate statusables: [Statusable],
+                          params: [String : Any],
+                          callName: String,
+                          delayInterval: TimeInterval = 2.0,
+                          viewsToIgnore: [UIView]) async throws -> Any {
+
+        // Trigger the loading event for all statusables
+        for statusable in statusables {
+            statusable.handleEvent(status: .loading)
+        }
+
+        // Reference the statusables weakly in case they are deallocated before the signal finishes.
+        let weakStatusables: [WeakAnyStatusable] = statusables.map { (statusable)  in
+            return WeakAnyStatusable(statusable)
+        }
+
+        return try await withCheckedThrowingContinuation({ (continuation: CheckedContinuation<Any, Error>) in
```

Comment:
This is where the magic happens. We're not in an async context so we have to create one here. "withCheckedThrowingContinuation" allows the following code to tell us when it's done or has encountered an error.